### ISSUE TITLE
Allow selection between warning, error and ignoring regarding the validity of links

### ISSUE THREAD

**Issue by santi100a** [remarkjs/remark-validate-links#77]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

It's kinda jarring and odd to use `--frail` to catch warnings as errors and exit with 1. Also, other Remark warnings could cause errors if they are raised at lint time.

### Solution

We should be able to specify the severity to be reported as "error", "warn" or "ignore".
Example:
```json
{
    "plugins": {
           "validate-links": {
                     "settings": {
                          "severity": "error" // also warn or ignore
                    }
           }
    }
}
```

### Alternatives

Another way would be to directly modify Remark itself, but I think this is much better.

**Comment by wooorm** [comment]
Continuing this conversation on from https://github.com/orgs/remarkjs/discussions/1217#discussioncomment-6796453:

What do you mean, “jarring”?

Why do you think broken URLs are errors?

What else are you using? Are you only using `remark-validate-links`?

>  It'd be nice if Remark could be adjusted like ESLint to bypass the severity of certain rules.

ESLint is just a linter.
remark isn’t just a linter. remark does many things: it can lint, but it also transforms.
On errors, it stops: because errors are fatal errors.
On warnings, it continues.
If this package would see broken URLs as errors, like you suggest, and there would be one of those broken URLs, *nothing after this package will run*. It stops. Because it’s an error. What nice about that?

---

From here:

> Also, other Remark warnings could cause errors if they are raised at lint time.

Can you expand on this scenario? What do you mean?

> Another way would be to directly modify Remark itself, but I think this is much better.

You mean the package [`remark`](https://github.com/remarkjs/remark/tree/main/packages/remark)? How would you change it?

---

Errors vs warnings in unified *mean* something. I don’t think you want the actual behavior that that difference results in.

**Comment by santi100a** [comment]
> > Also, other Remark warnings could cause errors if they are raised at lint time.
> 
> Can you expand on this scenario? What do you mean?
> 

What if I used another Remark plugin and it raises a warning? If I use `--frail`, Remark will fail even if I only want it to fail if `validate-links` warns, not other plugins.

**Comment by wooorm** [comment]
If you use plugins that raise regular messages, plugins run.
If one plugin uses a fatal message (an “error”), other plugins don’t run. Files are not written.

`--frail` is unrelated to this process. It exits with `1` if there was a regular *or* fatal message, instead of the normal `1` only for fatal messages.

**Comment by santi100a** [comment]
So the solution is to run Remark for link validation with `--frail` and run further linting without this flag.

**Comment by wooorm** [comment]
You never explained what your problem is, you proposed a solution.

I *think* the solution for your problem is to use `--frail`. And then you’re done?

**Comment by wooorm** [comment]
What you mention just now as a solution *would* be a solution to what you are proposing here though. To turn warnings for links into errors. Then you’d need to run twice.
If you don‘t turn warnings into errors, you don‘t need to run twice.

**Comment by wooorm** [comment]
Closing as I don’t yet think there is a good reason to add such code, `--frail` seems like a better solution.
